compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/Desugar.java

```diff
@@ -1365,16 +1369,18 @@ public void visit(BLangTupleVariable varNode) {
     @Override
     public void visit(BLangRecordVariable varNode) {
         final BLangBlockStmt blockStmt = ASTBuilderUtil.createBlockStmt(varNode.pos);
+        String name = String.format("$map$_%d$", recordVarCount++);
```

Comment:
Technically, these are also anon types, right? Shall we move this and other name generation logic to `BLangAnonymousModelHelper`?

Reply:
+1